### PR TITLE
Setup for local e2e development

### PR DESCRIPTION
**Pull request by jsugarman**
## Description of change
Document setup for e2e development

[Comes out of ticket](https://dsdmoj.atlassian.net/browse/CRM457-992)

### CODE REVIEW

File: docs/development-e2e-setup.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+- Submit a claim or prior authority application in the provider app
+- See a successful cron job being processed in the Sidekiq WebUI (at http://localhost/sidekiq).
+- See the successful authentication and processing in the app store server output.
+- See the submitted claim or application in the caseworker app list of claims or list of applications.
+
+
+## Trouble shooting
+
+- Missing translation errors in provider app
```
Comment (patrick-laa) [inline docs/development-e2e-setup.md:70]: This issue is now ticketed as CRM457-1183. Might be worth sticking that ticket number in this doc to make it more likely we remember to update this document to take the stale info out once that ticket is done.

File: docs/development-e2e-setup.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+SIDEKIQ_WEB_UI_USERNAME=sidekiq
+SIDEKIQ_WEB_UI_PASSWORD=sidekiq
+```
+
+## Running e2e development stack
+
+### Run the provider App:
+  - Follow above readme to setup the development requirements
+
+  - Amend Procfile.dev to run rails server on a unique port of your choice
```
Comment (patrick-laa) [inline docs/development-e2e-setup.md:36]: I'd almost recommend we hard-code 3001 into the procfile. If we're all going to have both caseworker and provider up and running simultaneously more and more, maybe we should all get into the habit of having one on 3001 and the other on 3002 or something, by having those as standard in the procfile.

**Review by patrick-laa — APPROVED** [review]
Feel free to ignore my suggestions - I'm not 100% convinced myself that they add benefit.

File: docs/development-e2e-setup.md (hunk 2 above)
Comment (jsugarman) [reply docs/development-e2e-setup.md:36]: i did muck about with `${PORT:-3000}` type logic but there are so many ways it seems to go wrong (puma config, foreman start command, Procfile.dev) that i have never found a robust solution, here or elsewhere.

So i would vote for hardcoding 3001/3002 into provider and caseworker 👍

File: docs/development-e2e-setup.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+```sh
+# App store credentials (for authenticating to a local app store currently)
+# TODO: Oauth may be removed from local/development app store in the future
+APP_STORE_CLIENT_ID=not-a-real-key
+APP_STORE_CLIENT_SECRET=not-a-real-key
+APP_STORE_TENANT_ID=not-a-real-key
+```
+Ask a dev where these can be retrieved, and do not use production versions
+
+To receive emails to you own email address, or avoid errors on submission, you should supply the following ENV VAR
```
Comment (jsugarman) [inline docs/development-e2e-setup.md:18]:
```suggestion
To receive emails to your own email address, or avoid errors on submission, you should supply the following ENV VAR
```

File: docs/development-e2e-setup.md (hunk 2 above)
Comment (jogilder) [reply docs/development-e2e-setup.md:36]: Agreed with this

**Pull request merged**